tests/unit/common/time/test_date.py

```diff
@@ -37,507 +36,512 @@
+        with pytest.raises(TypeError):
             _ = new_year - object()
 
     def test_date_less_than(self):
         new_year = Date(2000, 1, 1)
         christmas = Date(1999, 12, 25)
-        self.assertLess(christmas, new_year)
+        assert christmas < new_year
 
     def test_date_less_than_object(self):
         d = Date(2000, 1, 1)
-        with self.assertRaises(TypeError):
+        with pytest.raises(TypeError):
             _ = d < object()
 
     def test_date_less_than_or_equal_to(self):
         new_year = Date(2000, 1, 1)
         christmas = Date(1999, 12, 25)
-        self.assertLessEqual(christmas, new_year)
+        assert christmas <= new_year
 
     def test_date_less_than_or_equal_to_object(self):
         d = Date(2000, 1, 1)
-        with self.assertRaises(TypeError):
+        with pytest.raises(TypeError):
             _ = d <= object()
 
     def test_date_greater_than_or_equal_to(self):
         new_year = Date(2000, 1, 1)
         christmas = Date(1999, 12, 25)
-        self.assertGreaterEqual(new_year, christmas)
+        assert new_year >= christmas
 
     def test_date_greater_than_or_equal_to_object(self):
         d = Date(2000, 1, 1)
-        with self.assertRaises(TypeError):
+        with pytest.raises(TypeError):
             _ = d >= object()
 
     def test_date_greater_than(self):
         new_year = Date(2000, 1, 1)
         christmas = Date(1999, 12, 25)
-        self.assertGreater(new_year, christmas)
+        assert new_year > christmas
 
     def test_date_greater_than_object(self):
         d = Date(2000, 1, 1)
-        with self.assertRaises(TypeError):
+        with pytest.raises(TypeError):
             _ = d > object()
 
     def test_date_equal(self):
         d1 = Date(2000, 1, 1)
         d2 = Date(2000, 1, 1)
-        self.assertEqual(d1, d2)
+        assert d1 == d2
 
     def test_date_not_equal(self):
         d1 = Date(2000, 1, 1)
         d2 = Date(2000, 1, 2)
-        self.assertNotEqual(d1, d2)
+        assert d1 != d2
 
     def test_date_not_equal_to_object(self):
         d1 = Date(2000, 1, 1)
-        self.assertNotEqual(d1, object())
+        assert d1 != object()
 
-    def test_year_week_day(self):
-        for ordinal in range(Date(2001, 1, 1).to_ordinal(), Date(2008, 1, 1).to_ordinal()):
-            self.assertEqual(Date.from_ordinal(ordinal).iso_calendar(), date.fromordinal(ordinal).isocalendar())
+    @pytest.mark.parametrize("ordinal", (
+        Date(2001, 1, 1).to_ordinal(),
+        Date(2008, 1, 1).to_ordinal(),
+    ))
+    def test_year_week_day(self, ordinal):
+        assert Date.from_ordinal(ordinal).iso_calendar() \
+               == date.fromordinal(ordinal).isocalendar()
 
     def test_time_tuple(self):
         d = Date(2018, 4, 30)
-        self.assertEqual(d.time_tuple(), struct_time((2018, 4, 30, 0, 0, 0, 0, 120, -1)))
+        assert d.time_tuple() == struct_time(
+            (2018, 4, 30, 0, 0, 0, 0, 120, -1))
```

Comment:
Found a way to make it look pretty :)
```suggestion
        expected = struct_time((2018, 4, 30, 0, 0, 0, 0, 120, -1))
        assert d.time_tuple() == expected
```